api/v2/urls.go

```diff
@@ -43,8 +43,21 @@ func NewURLBuilderFromString(root string) (*URLBuilder, error) {
 // NewURLBuilderFromRequest uses information from an *http.Request to
 // construct the root url.
 func NewURLBuilderFromRequest(r *http.Request) *URLBuilder {
+	var scheme string
+
+	forwardedProto := r.Header.Get("X-Forwarded-Proto")
+
+	switch {
+	case len(forwardedProto) > 0:
+		scheme = forwardedProto
+	case len(r.URL.Scheme) > 0:
+		scheme = r.URL.Scheme
+	default:
+		scheme = "http"
+	}
+
 	u := &url.URL{
-		Scheme: r.URL.Scheme,
+		Scheme: scheme,
 		Host:   r.Host,
```

Comment:
Will `r.Host` be sufficient, as well? For example, do most proxies change the host header? If so, we may want to take this as configuration (that was the original plan).

Also, let's add a TODO here that uses incoming TLS configuration on request to detect scheme in absence of `X-Forwarded-Proto`.